support/metrics/metricsTracker.go

```diff
@@ -205,6 +207,7 @@ func MakeMetricsTracker(
 		botStartTime:        botStartTime,
 		isDisabled:          isDisabled,
 		updateEventSentTime: nil,
+		handler:             plugins.MakeTradeMetricsHandler(),
```

Comment:
@debnil we wouldn't construct this here -- it would get constructed outside here and passed in. the calling code would reuse this instance of tradeMetricsHandler to register it on the fill tracker which is where the HandleFill function will get invoked. does that make sense?